CMakeLists.txt

```diff
@@ -600,40 +613,37 @@ endif()
 if(NOT DEFINED Boost_INCLUDE_DIR OR NOT DEFINED Boost_LIBRARIES)
 	FIND_PACKAGE(Boost REQUIRED COMPONENTS system)
 endif()
-include_directories(${Boost_INCLUDE_DIRS})
-target_link_libraries(torrent-rasterbar ${Boost_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_include_directories(torrent-rasterbar PUBLIC ${Boost_INCLUDE_DIRS})
+target_link_libraries(torrent-rasterbar PUBLIC ${Boost_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
 
 if (WIN32)
-	target_link_libraries(torrent-rasterbar wsock32 ws2_32 Iphlpapi)
-	# target Windows Vista or later
-	target_compile_definitions(torrent-rasterbar PUBLIC _WIN32_WINNT=0x0600)
-	target_link_libraries(torrent-rasterbar debug dbghelp)
-	# prevent winsock1 to be included
-	target_compile_definitions(torrent-rasterbar PUBLIC WIN32_LEAN_AND_MEAN)
+	target_link_libraries(torrent-rasterbar
+		PRIVATE
+			wsock32 ws2_32 Iphlpapi
+			debug dbghelp
+	)
+
+	target_compile_definitions(torrent-rasterbar
+		PUBLIC
+			_WIN32_WINNT=0x0600 # target Windows Vista or later
+			WIN32_LEAN_AND_MEAN # prevent winsock1 to be included
```

Comment:
I recognise this wasn't introduced by this change, but does anyone have any thoughts on exporting this define? In one way it seems a bit intrusive. It's essentially saying no clients linking against libtorrent is allowed to use microsoft's macros. On the other hand, most likely only software written in the 90's use those anyway.

Reply:
Defining `WIN32_LEAN_AND_MEAN` has been SOP for 20 years, I doubt anyone is going to complain.